editor/src/clj/editor/code_completion.clj

```diff
@@ -147,6 +148,19 @@
   (s/keys :req-un [::insert-string]
           :opt-un [::tab-triggers ::exit-ranges :editor.code-completion.insert/cursor-range ::additional-edits]))
 
+(defn- ->ast [^ParseTree ctx rule-names]
+  (if (instance? ParserRuleContext ctx)
+    (let [^ParserRuleContext ctx ctx
+          node (cons (keyword (aget ^Object/1 rule-names (.getRuleIndex ctx)))
+                     (into []
+                           (map (fn [^long i]
+                                  (->ast (.getChild ctx i) rule-names)))
+                           (range (.getChildCount ctx))))]
+      (if (.-exception ctx)
+        (list ::error node)
+        node))
+    (.getText ctx)))
```

Comment:
This is a re-implementation of the clj-antler code that produces an AST. The intention with this PR is to reduce the editor build size, so I kept the behavior.